### PR TITLE
Gives Radiation storms sound

### PR DESCRIPTION
**Pull request by TheFakeElon**
## About The Pull Request

Gives radstorms a unique sound, made by yours truly

## Why It's Good For The Game

More sound, more fun

## Changelog
:cl:
add: Added "ambient" sound during radiation storms
/:cl:

### CODE REVIEW

**Comment by YakiAttaki** [comment]
modularise your code

**Comment by MCterra10** [comment]
oh god I'm MOOODULARIZING ARGHHHH

**Comment by MCterra10** [comment]
also @TheFakeElon did you make these entirely from scratch or did you sample sounds?

**Comment by TheFakeElon** [comment]
Stole a bunch of different sound effects off the internet, changed them up a lot in audacity and generated some white noise

**Comment by MCterra10** [comment]
did you check regarding licensing?

**Comment by TheFakeElon** [comment]
yes

File: code/datums/looping_sounds/weather.dm [hunk 1]
```diff
@@ -45,3 +45,13 @@
...
+
+/datum/looping_sound/rad_storm
+	mid_sounds = list(
+		'sound/weather/radstorm/radstorm1.ogg'=1,
+		)
+	mid_length = 80
+	start_sound = 'sound/weather/radstorm/radstorm_start.ogg'
+	start_length = 130
+	end_sound = 'sound/weather/radstorm/radstorm_end.ogg'
+	volume = 60
```
Comment (YakiAttaki) [inline code/datums/looping_sounds/weather.dm:57]: you need to put this under the austation folder, so something like `austation/code/datums/looping_sounds/weather.dm`. don't forget to put the path to this file in `includes.dm`

File: code/datums/weather/weather_types/radiation_storm.dm [hunk 2]
```diff
@@ -23,11 +23,40 @@
...
+	for (var/z in impacted_z_levels)
+		eligible_areas += SSmapping.areas_in_z["[z]"]
+	for(var/i in 1 to eligible_areas.len)
+		var/area/place = eligible_areas[i]
+		if(place.outdoors)
+			outside_areas += place
+		else
+			inside_areas += place
+		CHECK_TICK
+
```
Comment (YakiAttaki) [inline code/datums/weather/weather_types/radiation_storm.dm:43]: same here

Comment (YakiAttaki) [inline code/datums/weather/weather_types/radiation_storm.dm:43]: same with this here, minus line 29

File: code/datums/weather/weather_types/radiation_storm.dm [hunk 3]
```diff
@@ -23,11 +23,40 @@
...
+	. = ..()
+
+	radstorm1.start()
+
+/datum/weather/rad_storm/end()
+	. = ..()
+
+	radstorm1.stop()
+
+
```
Comment (YakiAttaki) [inline code/datums/weather/weather_types/radiation_storm.dm:59]: and here as well

File: code/datums/looping_sounds/weather.dm (hunk 1 above)
Comment (YakiAttaki) [inline code/datums/looping_sounds/weather.dm:57]: need to modularise these under the austation folder. make a file under same path under austation/ and just include your code changes, then put them into includes.dm to make sure they actually get used.

File: code/datums/weather/weather_types/radiation_storm.dm (hunk 2 above)
Comment (YakiAttaki) [inline code/datums/weather/weather_types/radiation_storm.dm:43]: same here

File: code/datums/weather/weather_types/radiation_storm.dm (hunk 3 above)
Comment (YakiAttaki) [inline code/datums/weather/weather_types/radiation_storm.dm:59]: and here too

**Comment by YakiAttaki** [comment]
also put the oggs under austation/sound/ whatever

**Comment by MCterra10** [comment]
yaki doing my job for me ahaha

**Comment by TheFakeElon** [comment]
based, I'll do it tonight or tomorrow, thanks :)

**Comment by MCterra10** [comment]
> 6 days later

**Comment by TheFakeElon** [comment]
Sorry, I can’t get it to work with modular code. Closing until I can fix it

**Comment by coolsurf6** [comment]
:(

**Comment by TheFakeElon** [comment]
I'm gonna try this shit again

**Comment by TheFakeElon** [comment]
with MODULAR code

**Comment by TheFakeElon** [comment]
wait, I broke a bunch of things, I'll make a new pr

**Comment by TheFakeElon** [comment]
And she was never heard from again.

I'll do this soon:tm:

**Pull request closed without merging**